stdlib/public/core/Reverse.swift

```diff
@@ -12,34 +12,47 @@
 
 /// An index that traverses the same positions as an underlying index,
 /// with inverted traversal direction.
-public struct ReverseIndex<Base> : Comparable {
-  public init(_ base: Base) { self.base = base }
```

Comment:
The question in my mind is who else would want to create instances of these. They are basically a layer of indirection that is generally specific to their related specialize collections. If they have utility to others I will restore the initializer. I have the same question about Filter, etc.